src/dune/dep.ml

```diff
@@ -1,9 +1,17 @@
 open Stdune
 
 module Trace = struct
+  module Dep = struct
```

Comment:
I think it would be nice to come up with another name for this because "dep" means so many things already. One idea is "Fact" because it records facts the action observed about the environment.